homeassistant/components/zha/logbook.py

```diff
@@ -0,0 +1,84 @@
+"""Describe ZHA logbook events."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import TYPE_CHECKING
+
+import voluptuous as vol
+
+from homeassistant.components.logbook.const import (
+    LOGBOOK_ENTRY_MESSAGE,
+    LOGBOOK_ENTRY_NAME,
+)
+from homeassistant.const import ATTR_DEVICE_ID
+from homeassistant.core import Event, HomeAssistant, callback
+import homeassistant.helpers.device_registry as dr
+
+from .core.const import DOMAIN as ZHA_DOMAIN, ZHA_EVENT
+from .core.helpers import async_get_zha_device
+
+if TYPE_CHECKING:
+    from .core.device import ZHADevice
+
+
+@callback
+def async_describe_events(
+    hass: HomeAssistant,
+    async_describe_event: Callable[[str, str, Callable[[Event], dict[str, str]]], None],
+) -> None:
+    """Describe logbook events."""
+    device_registry = dr.async_get(hass)
+
+    @callback
+    def async_describe_zha_event(event: Event) -> dict[str, str]:
+        """Describe zha logbook event."""
+        device: dr.DeviceEntry | None = None
+        device_name: str = "Unknown device"
+        zha_device: ZHADevice | None = None
+        event_data: dict = event.data
+        event_type: str | None = None
+        event_subtype: str | None = None
+
+        try:
+            device = device_registry.devices[event.data[ATTR_DEVICE_ID]]
+            if device:
+                device_name = device.name_by_user or device.name or "Unknown device"
+            zha_device = async_get_zha_device(hass, event.data[ATTR_DEVICE_ID])
+        except (KeyError, AttributeError):
+            pass
+
+        if zha_device and zha_device.device_automation_triggers:
+            for (
+                etype,
+                subtype,
+            ), trigger in zha_device.device_automation_triggers.items():
+                event_data_schema = vol.Schema(
+                    {vol.Required(key): value for key, value in trigger.items()},
+                    extra=vol.ALLOW_EXTRA,
+                )
+                try:
+                    if event_data_schema:
+                        event_data_schema(event_data)
+                    event_type = etype
+                    event_subtype = subtype
+                    break
+                except vol.Invalid:
+                    # If event doesn't match, skip event
+                    continue
```

Comment:
Is there a way we can avoid this linear search?  This is going to get called for every zha row.  If they have a ton of events we potentially have a large loop inside of a large loop which could be a scale issue

Reply:
I could drop the translation to the device trigger information? I think that would be slightly confusing to the users though. The problem is that I am using the trigger schemas to determine if there is a match. I may have an idea that could make it a bit better. Stay tuned.

Reply:
Thanks again for the assist!